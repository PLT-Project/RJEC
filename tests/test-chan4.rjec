func foo1 (r chan int, s chan int) {
    for {
        a := <- r;
        prints("receiver:");
        printi(a + 1);
        s <- (a * 2);
    }
}

func main () {
    s, r := make(chan int), make(chan int);
    yeet foo1(s, r);
    for a := 1; a < 1000 ; a = (<- r) {
        s <- a;
        prints("sender:");
        printi(a);
    }
}