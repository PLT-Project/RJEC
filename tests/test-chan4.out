receiver:
2
sender:
1
receiver:
3
sender:
2
receiver:
5
sender:
4
receiver:
9
sender:
8
receiver:
17
sender:
16
receiver:
33
sender:
32
receiver:
65
sender:
64
receiver:
129
sender:
128
receiver:
257
sender:
256
receiver:
513
sender:
512